pyexchange/coinone.py

```diff
@@ -0,0 +1,256 @@
+        return list(map(lambda item: CoinoneOrder.from_message(item, pair), orders["limitOrders"]))
+
+    # Calculate the asking price units for a given price range
+    def _calc_price_precision(self, price: float) -> int:
+        if 1000 <= price < 5000:
+            return 1
+        elif 5000 <= price < 10000:
+            return 5
+        elif 10000 <= price < 50000:
+            return 10
+        elif 50000 <= price < 100000:
+            return 50
+        elif 100000 <= price < 500000:
+            return 100
+        elif 500000 <= price < 1000000:
+            return 500
+        elif 1000000 <= price:
+            return 1000
+
+    def place_order(self, pair: str, is_sell: bool, price: Wad, amount: Wad) -> str:
+        assert (isinstance(pair, str))
+        assert (isinstance(is_sell, bool))
+        assert (isinstance(price, Wad))
+        assert (isinstance(amount, Wad))
+
+        side = "limit_buy" if is_sell is False else "limit_sell"
+        currency = pair.split('-')[0]
+
+        # Coinone krw price precision must be specified in thousands or less
+        float_price = Wad.__float__(price)
+        price_prec = self._calc_price_precision(float_price)
+        price = round(float_price / price_prec, 0) * price_prec
+
+        data = {
+            "currency": currency,
+            "price": str(price),
+            "qty": str(round(Wad.__float__(amount), 2))
+        }
+
+        self.logger.info(f"Placing order ({side}, amount {data['qty']} of {pair},"
+                         f" price {data['price']})...")
+        response = self._http_authenticated_request("POST", f"/v2/order/{side}", data)
+
+        order_id = ""
+
+        if response['result'] == 'success':
+            order_id = response['orderId']
+            self.logger.info(f"Placed order (#{order_id})")
+
+        return order_id
+
+    def cancel_order(self, order_id: str, pair: str, price: Wad, amount: Wad, is_sell: bool) -> bool:
+        assert (isinstance(order_id, str))
+        assert (isinstance(pair, str))
+        assert (isinstance(is_sell, bool))
+        assert (isinstance(price, Wad))
+        assert (isinstance(amount, Wad))
+
+        self.logger.info(f"Cancelling order #{order_id}...")
+
+        currency = pair.split('-')[0]
+        is_ask = 1 if is_sell is True else 0
+
+        data = {
+            "order_id": order_id,
+            "currency": currency,
+            "price": str(round(Wad.__float__(price), 2)),  # quote token is always krw
+            "qty": str(round(Wad.__float__(amount), 2)),
+            "is_ask": is_ask
+        }
+
+        result = self._http_authenticated_request("POST", f"/v2/order/cancel", data)
+        return True if result["result"] == "success" else False
+
+    def get_trades(self, pair: str, offset: int = 0) -> List[Trade]:
+        assert (isinstance(pair, str))
+        assert (isinstance(offset, int))
+
+        currency = pair.split('-')[0]
```

Comment:
Not sure I understand this.  Why would we only request trades for the quote token?  Wouldn't that include pairs for other base currencies?

Reply:
The quote currency is KRW for every book, so their API just takes the base currency ETH or DAI, for example, and returns the trades for that currency assumed to be quoted in KRW.